rust/src/operations/write.rs

```diff
@@ -0,0 +1,466 @@
+//! Used to write a [RecordBatch] into a delta table.
+//!
+//! New Table Semantics
+//!  - The schema of the [RecordBatch] is used to initialize the table.
+//!  - The partition columns will be used to partition the table.
+//!
+//! Existing Table Semantics
+//!  - The save mode will control how existing data is handled (i.e. overwrite, append, etc)
+//!  - The schema of the RecordBatch will be checked and if there are new columns present
+//!    they will be added to the tables schema. Conflicting columns (i.e. a INT, and a STRING)
+//!    will result in an exception
+//!  - The partition columns, if present are validated against the existing metadata. If not
```

Comment:
Adding some missing punctuation:

```suggestion
//!    will result in an exception.
//!  - The partition columns, if present, are validated against the existing metadata. If not
```